### PR TITLE
Add GoImports Post Processor

### PR DESCRIPTION
**Pull request by asahasrabuddhe**
Thanks a lot for this wonderful framework. I was building a protoc-gen plugin and needed a post processor for `GoImports`. As it was not a part of this framework, I implemented my own. I think that this could be a useful part of this framework.

### CODE REVIEW

**Comment by upils** [comment]
In the `Match()` method, setting `g.filename` is not ineffective since the method is not working on a pointer.

But this is never use by `Process()` so we rewrote it this way:


```go
type goImports struct {}

// GoImports returns a PostProcessor that run goimports on any files ending . ".go"
func GoImports() pgs.PostProcessor { return goImports{} }

func (g goImports) Match(a pgs.Artifact) bool {
	var n string

	switch a := a.(type) {
	case pgs.GeneratorFile:
		n = a.Name
	case pgs.GeneratorTemplateFile:
		n = a.Name
	case pgs.CustomFile:
		n = a.Name
	case pgs.CustomTemplateFile:
		n = a.Name
	default:
		return false
	}

	return strings.HasSuffix(n, ".go")
}

func (g goImports) Process(in []byte) ([]byte, error) {
	// We do not want to give a filename here, ever
	return imports.Process("", in, nil)
}

```

**Comment by asahasrabuddhe** [comment]
thanks @upils ! I have updated my branch with your feedback.

@keith @ardakuyumcu @pdecks Could you please review this? Thanks!

File: lang/go/goimports_test.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import (
+	pgs "github.com/lyft/protoc-gen-star"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestGoImports_Match(t *testing.T) {
+	t.Parallel()
+
+	pp := GoFmt()
```
Comment (pdecks) [inline lang/go/goimports_test.go:12]: Should this be `GoImports()`?

File: lang/go/goimports.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+		return false
+	}
+
+	return strings.HasSuffix(n, ".go")
+}
+
+func (g goImports) Process(in []byte) ([]byte, error) {
+	// We do not want to give a filename here, ever
+	return imports.Process("", in, nil)
+}
```
Comment (pdecks) [inline lang/go/goimports.go:36]: Match https://github.com/lyft/protoc-gen-star/blob/master/lang/go/gofmt.go#L34

```suggestion
func (p goFmt) Process(in []byte) ([]byte, error) { return format.Source(in) }
```

File: lang/go/goimports.go [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+// GoImports returns a PostProcessor that run goimports on any files ending . ".go"
+func GoImports() pgs.PostProcessor { return goImports{} }
+
+func (g goImports) Match(a pgs.Artifact) bool {
+	var n string
+
+	switch a := a.(type) {
+	case pgs.GeneratorFile:
+		n = a.Name
```
Comment (pdecks) [inline lang/go/goimports.go:19]:
```suggestion
	n = a.Name
```

File: lang/go/goimports.go [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+	case pgs.GeneratorTemplateFile:
+		n = a.Name
```
Comment (pdecks) [inline lang/go/goimports.go:21]:
```suggestion
	n = a.Name
```

File: lang/go/goimports.go [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+	case pgs.CustomFile:
+		n = a.Name
```
Comment (pdecks) [inline lang/go/goimports.go:23]:
```suggestion
	n = a.Name
```

File: lang/go/goimports.go [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+	case pgs.CustomTemplateFile:
+		n = a.Name
```
Comment (pdecks) [inline lang/go/goimports.go:25]:
```suggestion
	n = a.Name
```

File: lang/go/goimports.go [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+	default:
+		return false
```
Comment (pdecks) [inline lang/go/goimports.go:27]:
```suggestion
	return false
```

File: lang/go/goimports_test.go (hunk 1 above)
Comment (pdecks) [inline lang/go/goimports_test.go:12]:
```suggestion
	pp := GoImports()
```

File: lang/go/goimports.go (hunk 2 above)
Comment (pdecks) [inline lang/go/goimports.go:36]: Match https://github.com/lyft/protoc-gen-star/blob/master/lang/go/gofmt.go#L36

```suggestion
}

var _ pgs.PostProcessor = goImports{}
```

**Comment by asahasrabuddhe** [comment]
Hello @pdecks I have updated the PR with your feedback.

**Comment by pdecks** [comment]
It looks like the [linter test](https://github.com/lyft/protoc-gen-star/runs/7847875914?check_suite_focus=true) started failing after this merge commit: https://github.com/lyft/protoc-gen-star/pull/96/commits/cd567a8de02cdc28453b8b44eaba7f50175748f2

**Comment by asahasrabuddhe** [comment]
> It looks like the [linter test](https://github.com/lyft/protoc-gen-star/runs/7847875914?check_suite_focus=true) started failing after this merge commit: [cd567a8](https://github.com/lyft/protoc-gen-star/commit/cd567a8de02cdc28453b8b44eaba7f50175748f2)

This is now fixed @pdecks

**Review by pdecks — APPROVED** [review]

**Comment by asahasrabuddhe** [comment]
@pdecks When can we expect a new release with this PR?

**Comment by pdecks** [comment]
> @pdecks When can we expect a new release with this PR?

You'll need to address the error in the `pre-commit` check first.

**Comment by asahasrabuddhe** [comment]
> > @pdecks When can we expect a new release with this PR?
> 
> You'll need to address the error in the `pre-commit` check first.

I've fixed the `pre-commit` check but now the `pre-commit.ci` is failing. I have looked at the error and it does not look to be failing due to the code. There's some connection timeout error there.

**Comment by pdecks** [comment]
> > > @pdecks When can we expect a new release with this PR?
> > 
> > 
> > You'll need to address the error in the `pre-commit` check first.
> 
> I've fixed the `pre-commit` check but now the `pre-commit.ci` is failing. I have looked at the error and it does not look to be failing due to the code. There's some connection timeout error there.

Thanks for the heads up -- a teammate had enabled `pre-commit.ci` right after I merged the previous commit, and it looks like we're running into something akin to this issue https://github.com/pre-commit-ci/issues/issues/29. We'll disable it for now.

**Review by jcapote-lyft — APPROVED** [review]

**Pull request merged**